extensions/pkg/controller/healthcheck/reconciler.go

```diff
@@ -144,65 +146,75 @@ func (r *reconciler) performHealthCheck(ctx context.Context, request reconcile.R
 
 	for _, healthCheckResult := range *healthCheckResults {
 		// get or init conditions on extension resource
-		healthCondition := gardencorev1beta1helper.GetOrInitCondition(r.registeredExtension.extension.GetExtensionStatus().GetConditions(), gardencorev1beta1.ConditionType(healthCheckResult.HealthConditionType))
-		if !healthCheckResult.IsHealthy && healthCheckResult.FailedChecks > 0 {
-			r.logger.Info("Updating HealthCheckCondition for extension resource to ConditionCheckError.", "kind", r.registeredExtension.groupVersionKind.Kind, "health condition type", healthCheckResult.HealthConditionType, "name", request.Name, "namespace", request.Namespace)
-			if err := r.updateExtensionConditionToConditionCheckError(ctx, r.registeredExtension.extension, extension, healthCondition, r.registeredExtension.groupVersionKind.Kind, healthCheckResult); err != nil {
-				return r.resultWithRequeue(), err
+		healthCondition := gardencorev1beta1helper.GetOrInitCondition(extension.GetExtensionStatus().GetConditions(), gardencorev1beta1.ConditionType(healthCheckResult.HealthConditionType))
+
+		if healthCheckResult.Status == gardencorev1beta1.ConditionProgressing || healthCheckResult.Status == gardencorev1beta1.ConditionFalse {
+			if healthCheckResult.FailedChecks > 0 {
+				r.logger.Info("Updating HealthCheckCondition for extension resource to ConditionCheckError.", "kind", r.registeredExtension.groupVersionKind.Kind, "health condition type", healthCheckResult.HealthConditionType, "name", request.Name, "namespace", request.Namespace)
+				if err := r.updateExtensionConditionToConditionCheckError(ctx, extension, healthCondition, r.registeredExtension.groupVersionKind.Kind, healthCheckResult); err != nil {
+					return r.resultWithRequeue(), err
+				}
+				continue
 			}
-			continue
-		}
 
-		if !healthCheckResult.IsHealthy {
-			r.logger.Info("Health check for extension resource unsuccessful.", "kind", fmt.Sprintf("%s.%s.%s", r.registeredExtension.groupVersionKind.Kind, r.registeredExtension.groupVersionKind.Group, r.registeredExtension.groupVersionKind.Version), "name", request.Name, "namespace", request.Namespace, "failed", healthCheckResult.FailedChecks, "successful", healthCheckResult.SuccessfulChecks, "details", healthCheckResult.GetDetails())
-			if err := r.updateExtensionConditionToError(ctx, r.registeredExtension.extension, extension, healthCondition, healthCheckResult); err != nil {
+			r.logger.Info("Health check for extension resource progressing or unsuccessful.", "kind", fmt.Sprintf("%s.%s.%s", r.registeredExtension.groupVersionKind.Kind, r.registeredExtension.groupVersionKind.Group, r.registeredExtension.groupVersionKind.Version), "name", request.Name, "namespace", request.Namespace, "failed", healthCheckResult.FailedChecks, "progressing", healthCheckResult.ProgressingChecks, "successful", healthCheckResult.SuccessfulChecks, "details", healthCheckResult.GetDetails())
```

Comment:
This might lead to a lot of log messages in case many cluster are up and down scaling. Maybe it could avoid logging on info level when it is Progressing?

Reply:
done